scripts/deleteKey.js

```diff
@@ -1,38 +1,51 @@
 require('dotenv').config()
-const db = require('../db')
+const db = require('../src/db')
 const log = require('fancy-log')
 const logSymbols = require('log-symbols')
 const readline = require('readline')
 
-const rl = readline.createInterface({
-    input: process.stdin,
-    output: process.stdout
-  });
-
-const key = process.argv[process.argv.length - 1];
-
-db.get(key, (err) => {
+const handleDel = (key, cb) => (err)=>{
     if(err) {
+        console.error(logSymbols.error, 'Error deleting the key' + key)
+        console.error(JSON.stringify(err))
+        return cb({error: 'cannot be deleted'});
+    }
+    console.log(logSymbols.success, `Key ${key} deleted `)
+    return cb({success: true})
+}
+
+const handleGet = (key, cb, rl) => (err, value) => {
+    if(err !== null || typeof value === 'undefined') {
         console.error(logSymbols.error, 'Error key ' + key + ' not found')
         rl.close();
-        return;
+        return cb({error: 'not found'});
     }
-    rl.question(`Are you sure you xwant to remove the key ${key}? [y/n]`, (answer) => {
+    rl.question(`Are you sure you want to remove the key ${key}? [y/n]`, (answer) => {
         rl.close()
 
         if(answer.toLowerCase() !== 'y') {
-            return
+            return cb({cancel: true})
         }
 
-        db.del(key, (err)=>{
-            if(err) {
-                console.error(logSymbols.error, 'Error deleting the key' + key)
-                console.error(JSON.stringify(err))
-                return;
-            }
-            console.log(logSymbols.success, `Key ${key} deleted `)
-            console.log(key)
-            return
-        })     
+        db.del(key, handleDel(key, cb))
     })
-})
\ No newline at end of file
+}
+
+const deleteKey = (key, cb) => {
+    const rl = readline.createInterface({
+        input: process.stdin,
+        output: process.stdout
+    });
+    db.get(key, handleGet(key, cb, rl))
+}
+
+if(process.env.NODE_ENV !== 'test') {
+    const key = process.argv[process.argv.length - 1];
+    deleteKey(key, ()=>{})
+}
+
+module.exports = {
+    deleteKey,
+    handleGet,
```

Comment:
These exports are not being imported anywhere beside the test file.
To test the script I think we should export just the main function: delete key and perform all tests over it, otherwise, we may being testing implementation rather than functionality.
handleGet and handleDel looks like internal functions of this module to me.
I guess that it isn't trivial to write test in this proposed way anyway. May we iterate a solution together?